src/main/java/microsoft/exchange/webservices/data/core/request/ServiceRequestBase.java

```diff
@@ -632,16 +639,30 @@ protected SoapFaultDetails readSoapFault(EwsServiceXmlReader reader) {
   protected HttpWebRequest validateAndEmitRequest() throws Exception {
     this.validate();
 
-    HttpWebRequest request = this.buildEwsHttpWebRequest();
+    final HttpWebRequest request = this.buildEwsHttpWebRequest();
 
     try {
+      Callable<Object> task = new Callable<Object>() {
```

Comment:
`Callable` is a functional interface, so I think you can use a lambda here:
```java
Callable<Object> task = () -> getEwsHttpWebResponse(request);
```

Reply:
We're using Java 1.6, which doesn't support lambdas. Not sure if there's a specific reason, so I'll keep it for now.